### DISCUSSION TITLE
Change hostname and mutipass / ICS don't work anymore?

### DISCUSSION THREAD

**Discussion by jfmatth** [canonical/multipass#3652] [Q&A]
Very strange behavior.

Windows 11 Pro, Hyper-V, Multipass 1.14 (and older ones), Ubuntu 22.04

Installs just fine, apt-update works great, multiple reboots.

Hyper-V seems to want to change the IP one each reboot, which is fine each time, **until**....

I change the hostname to something with the suffix .mshome.net so I can reach it from Windows.  After I do that, nothing updates?

Anyone ever seen this, or what am I doing wrong or other ways to get around.

Ultimately, I'm just trying to use K3s and be able to have the local name use the ingress with the <hostname>.mshome.net name 

Very odd.

**Comment by georgeliao** [comment]
Hi @jfmatth 

Thanks for reporting. It is generally not recommended to use .mshome.net for custom hostnames due to potential conflicts with Windows' networking and DNS services. Why would you want to do that?

**Reply by jfmatth** [reply]
hi @georgeliao  Happy Friday :)

I love MP and using it with Hyper-V, I've even made a simple batch file that allows me to build certain sized VM's and their names.  simple, but effective for me 

I want to start using my laptop with MP and HV for Kubernetes testing against K3s.  I was thinking that to use the ingress on K8's, i'd need a FQDN that resolved on all ends, but maybe that isn't the case?

I tried more last night to figure out how to make this work, and I honestly don't think it's a MP issue, but a HV / ICS issue.  I messed with the cloud-init templates, etc.  Nothing seems to help.

I'm going to leave it alone at this point and see if K3s ingress works as is.  I can resolve from Windows to the URL, but not sure how the ingress will react.

**Reply by jfmatth** [reply]
> Hi @jfmatth
> 
> Thanks for reporting. It is generally not recommended to use .mshome.net for custom hostnames due to potential conflicts with Windows' networking and DNS services. Why would you want to do that?

BTW @georgeliao Kudo's on the 1.14 release, a VERY cool UI and nice to use.  Thanks for helping move MP forward, I've always thought this was a great tool.

Windows people are always amazed when I can demo WSL+HV+MP.
